### PR TITLE
MapVote+ (no actual mapping changes)

### PR DESCRIPTION
**Pull request by covertcorvid**
## About The Pull Request
This PR adds a new UI for displaying map/ship information during the map vote. The details include some lore, key features of each ship, and some fun statistics. This will help inform players about differences between the maps when they are choosing.
It also adds some logging on the server side and gives the admins the option to block users from mapvoting if they are believed to be voting in bad faith.

## Why It's Good For The Game
Lets people know what they're voting for even if not familiar with all maps

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

## Changelog
:cl:
add: Added new mapvote UI that displays detailed information about ships
add: Added additional ship data to all the map JSONs
add: Added statistics for round results and engine meltdowns
admin: New ban type "Mapvote" prevents people from voting in map votes
server: Logged who voted for what map and whether voters were present next round during the roundstart logout report in the Vote log
/:cl:

### CODE REVIEW

File: _maps/shrike.json [hunk 1]
```diff
@@ -11,5 +11,18 @@
 	    "emergency": "emergency_void"},
     "mine_disable": 1,
 	"ship_type": "/obj/structure/overmap/nanotrasen/missile_cruiser/starter/shrike",
-	"starmap_path": "config/starmap/starmap.json"
+	"starmap_path": "config/starmap/starmap.json",
+
+	"map_description": "The Shrike was once in service some time ago but has recently been unearthed as a schematic for cheap, fast ships perfect for use as a QRF. Most if not all Shrike-Class ships have been abandoned for some time in one shipyard or another until Skink Industries bought them en masse and outfitted them to the bare minimum requirements. As a result, Shrike vessels lack some amenities and technologies though with plenty of room for crew improvements.",
+	"manufacturer": "Skink Industries",
```
Comment (Karmic-Skink) [inline _maps/shrike.json:17]: Are we really going with the shameless self-insert because I couldn't remember any of the in-lore companies?

Comment (covertcorvid) [reply _maps/shrike.json:17]: I mostly just did whatever was in the document, feel free to suggest stuff or just tell me to look at the wiki lol

Comment (covertcorvid) [reply _maps/shrike.json:17]: I don't care if you just check it out and push changes either but also I'll sort comments out when I get home from pinball

Comment (covertcorvid) [inline _maps/shrike.json:17]:
```suggestion
	"manufacturer": "FrozenStar Armory",
```

File: _maps/shrike.json [hunk 2]
```diff
@@ -10,6 +10,19 @@
 		"ferry": "ferry_kilo",
 	    "emergency": "emergency_void"},
     "mine_disable": 1,
-	"ship_type": "/obj/structure/overmap/nanotrasen/missile_cruiser/starter/shrike",
-	"starmap_path": "config/starmap/starmap.json"
+	"ship_type": "/obj/structure/overmap/nanotrasen/frigate/starter/shrike",
+	"starmap_path": "config/starmap/starmap.json",
+
+	"map_description": "The Shrike was once in service some time ago but has recently been unearthed as a schematic for cheap, fast ships perfect for use as a QRF. Most if not all Shrike-Class ships have been abandoned for some time in one shipyard or another until Skink Industries bought them en masse and outfitted them to the bare minimum requirements. As a result, Shrike vessels lack some amenities and technologies though with plenty of room for crew improvements.",
```
[the remark this answers is not in the document]
Comment (Karmic-Skink) [reply _maps/shrike.json:16]: Definitely needs to also lose "QRF" as its an unexplained "TLA" that requires prior knowledge.

File: _maps/atlas.json [hunk 3]
```diff
@@ -25,6 +25,21 @@
         "emergency": "emergency_atlas"
     },
     "mine_disable": 1,
-    "ship_type": "/obj/structure/overmap/nanotrasen/missile_cruiser/starter",
-	"starmap_path": "config/starmap/starmap.json"
+    "ship_type": "/obj/structure/overmap/nanotrasen/frigate/starter",
```
Comment (benbot16) [inline _maps/atlas.json:28]:
```suggestion
    "ship_type": "/obj/structure/overmap/nanotrasen/missile_cruiser/starter",
```
/obj/structure/overmap/nanotrasen/frigate/starter doesn't exist.

Comment (covertcorvid) [reply _maps/atlas.json:28]: I made it exist. See the nanotrason.dm changes.

Comment (benbot16) [reply _maps/atlas.json:28]: I saw that, it's just that the atlas's overmap broke on live, so something might've gone wrong with that.

File: _maps/shrike.json [hunk 4]
```diff
@@ -10,6 +10,19 @@
 		"ferry": "ferry_kilo",
 	    "emergency": "emergency_void"},
     "mine_disable": 1,
-	"ship_type": "/obj/structure/overmap/nanotrasen/missile_cruiser/starter/shrike",
-	"starmap_path": "config/starmap/starmap.json"
+	"ship_type": "/obj/structure/overmap/nanotrasen/frigate/starter/shrike",
+	"starmap_path": "config/starmap/starmap.json",
+
+	"map_description": "The Shrike was once in service some time ago but has recently been unearthed as a schematic for cheap, fast ships perfect for use as a QRF. Most if not all Shrike-Class ships have been abandoned for some time in one shipyard or another until Whiterapids bought them en masse and outfitted them to the bare minimum requirements. As a result, Shrike vessels lack some amenities and technologies, though they leave plenty of room for crew improvements.",
```
Comment (Karmic-Skink) [inline _maps/shrike.json:16]:
```suggestion
	"map_description": "Shrike-class vessels were initially commissioned last century and served as light strike frigates before being retired to deep space shipyards in favour of more durable designs. Due to escalating conflict with the Syndicate, WhiteRapids looked to retrofitting decommissioned Shrike-class vessels as cheap alternative to production of brand-new warships. As a result, several more modern technologies are incompatible with the ageing hulls.",
```

Comment (Karmic-Skink) [reply _maps/shrike.json:16]: Or something to that effect.

File: _maps/shrike.json [hunk 5]
```diff
@@ -10,6 +10,19 @@
...
+	"manufacturer": "FrozenStar Armory",
+	"mapper": "Karmic-Skink",
+	"commission_date": "May 23, 2261",
```
Comment (Karmic-Skink) [inline _maps/shrike.json:19]:
```suggestion
	"commission_date": "May 23, 2175",
```
Re: discussion regarding this feature

**Review by Bokkiewokkie — DISMISSED** [review]

**Review by Bokkiewokkie — APPROVED** [review]

**Pull request merged**